### PR TITLE
Core/TempSummons: avoid use MoveInLineOfSight before InitSummon

### PR DESCRIPTION
**Pull request by EricksOliveira**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/18882
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15911

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Enter ICC and go to the Boss Lich King
2. I reached Phase 3 of the Lich King
3. Vile Spirits do not descend to attack Pets and Totems.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by skelUA** [comment]
Hi
What should  this PR  change in gane ?

**Comment by EricksOliveira** [comment]
Fixes issues like Creatures attacking Pet/Totem first. Mainly the Bug that occurs in ICC with Vile Spirits. Also fixes the problem with Peimeiro Raging Spirit appearing too quickly and falling off the Platform.

**Comment by tabooo** [comment]
I have tested and it works.

**Comment by Grimdhex** [comment]
@EricksOliveira could you update your pr and address cli/reviews ?

File: src/server/game/Entities/Object/Object.h [hunk 1]
```diff
@@ -553,6 +553,7 @@ class WorldObject : public Object, public WorldLocation
 
     void DestroyForNearbyPlayers();
     virtual void UpdateObjectVisibility(bool forced = true, bool fromUpdate = false);
+	virtual void UpdateObjectVisibilityOnCreate() { UpdateObjectVisibility(true); }
```
Comment (Grimdhex) [inline src/server/game/Entities/Object/Object.h:556]: I don't see the utility of this method ? Why not just use UpdateObjectVisibility(true) ?

Comment (Grimdhex) [reply src/server/game/Entities/Object/Object.h:556]: I prefer to directly use UpdateObjectVisibility(true) and just add a description of why we do this. Although it's understandable, but technically you're adding extra instructions, work for the compiler, and lengthening the list of methods available on completion tools.

Comment (EricksOliveira) [reply src/server/game/Entities/Object/Object.h:556]: Thanks for the feedback! I understand your point, and it makes sense to simplify by directly using UpdateObjectVisibility(true). I'll update the PR to remove the UpdateObjectVisibilityOnCreate method and include a descriptive comment explaining why this call is made. Let me know if there’s anything else you’d like me to adjust!

Comment (EricksOliveira) [reply src/server/game/Entities/Object/Object.h:556]: I tried some fixes, but it generates some unwanted errors, especially in Modules. I believe this is ideal. If possible, you can do the Merge.

**Pull request closed without merging**